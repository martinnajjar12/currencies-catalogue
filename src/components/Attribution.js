import { Link } from '@material-ui/core';
import React from 'react';

const Attribution = () => (
  <Link href="https://nomics.com">Crypto Market Cap & Pricing Data Provided By Nomics</Link>
);

export default Attribution;
