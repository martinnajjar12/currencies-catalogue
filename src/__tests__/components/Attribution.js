import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import Attribution from '../../components/Attribution';

describe('Attribution component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Attribution />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a link that its href attribute value is https://nomics.com', () => {
    const wrapper = shallow(<Attribution />);
    const link = wrapper.find('.makeStyles-subtitle1-2');
    expect(link.html()).toEqual('<a class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-subtitle1-2 MuiTypography-colorPrimary" target="_blank" href="https://nomics.com">CRYPTO MARKET CAP &amp; PRICING DATA PROVIDED BY NOMICS</a>');
  });
});
