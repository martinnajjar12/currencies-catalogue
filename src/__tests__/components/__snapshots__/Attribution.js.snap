// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribution component should match the snapshot 1`] = `
<div
  className="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-subtitle1-2 MuiTypography-colorPrimary"
    href="https://nomics.com"
    onBlur={[Function]}
    onFocus={[Function]}
    target="_blank"
  >
    CRYPTO MARKET CAP & PRICING DATA PROVIDED BY NOMICS
  </a>
</div>
`;
